lib/generators/boring/webmock/install/install_generator.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Boring
+  module Webmock
+    class InstallGenerator < Rails::Generators::Base
+      desc "Adds webmock gem to the application"
+
+      SUPPORTED_TEST_FRAMEWORKS = %w[rspec minitest]
+
+      class_option :test_gem, type: :string,
+                   desc: "Tell us the framework you use for writing tests in your application. Supported options are #{SUPPORTED_TEST_FRAMEWORKS}"
+      
+      def verify_presence_of_supported_test_gem
+        test_gem = options[:test_gem]
+
+        raise(NotImplementedError, "test_gem option can't be blank! Supported test_gem options are #{SUPPORTED_TEST_FRAMEWORKS}") if test_gem.blank?
+
+        if SUPPORTED_TEST_FRAMEWORKS.exclude?(test_gem)
+          raise(NotImplementedError, "#{test_gem} is not supported as a test_gem option! Supported options are #{SUPPORTED_TEST_FRAMEWORKS}")
+        end
+
+        if test_gem.eql?('rspec')
+          verify_presence_of_test_gem('rspec-rails')
+        else
+          verify_presence_of_test_gem('minitest')
+        end
+      end
+
+      def add_webmock_gem
```

Comment:
I think we should allow adding the gem regardless of requirement. Maybe make this the first step? If the framework configuration do not meet requirement, we can raise the error.

Reply:
If we will eventually raise error for test framework related conditions, isn't it good to raise error in the beginning so users know what to configure and run the generator again instead of them ending up with configuration that will not work?